src/services/auth/plugins/roles/preHandlers.ts

```diff
@@ -0,0 +1,25 @@
+import { FastifyReply, FastifyRequest, RouteHandlerMethod } from 'fastify';
+
+import { InsufficientPermission } from '../../error';
+import { RoleStrategy } from './types';
+
+/**
+ * preHandler to check if user validate one of the preconditions to access a route.
+ * @param strategies One or more role strategies to check if user validate the preconditions.
+ * @throws {InsufficientPermission} If user does not validate any of the preconditions.
+ * @throws {GraaspAuthError} If only one role strategy is provided and it failed with a provided error.
+ */
+export function whitelistRoles(...strategies: RoleStrategy[]): RouteHandlerMethod {
```

Comment:
I'm not sure if this name really help to understand what the function does, I don't found a better name for now but it could be something like that `authorizeUserWithAtLeastOnePassingStrategy`. It's too long I know ! But it adds clarity... I'll try to suggest something shorter ^^ What do you think ? @Defferrard , @spaenleh and @pyphilia ^^ If you think this name is good, just skip my comment :)